### PR TITLE
assign ipv6 address to aws

### PR DESCRIPTION
**Pull request by PratikDeoghare**
**What this PR does / why we need it**:
Assigns IPv6 address to a node if the pod cidr demands it. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged)*:
Implements part of https://github.com/kubermatic/kubermatic/issues/9294

**Optional Release Note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by moadqassem** [comment]
@PratikDeoghare PTAL here https://github.com/kubermatic/machine-controller/pull/1207#pullrequestreview-915002216

File: pkg/cloudprovider/types/types.go [hunk 1]
```diff
@@ -81,6 +82,32 @@ type NetworkConfig struct {
...
 }
 
+type IPVersion int
+
+const (
+	IPv4 = iota
+	IPv6
+)
+
+func (n NetworkConfig) ContainsCIDR(version IPVersion) bool {
```
Comment (moadqassem) [inline pkg/cloudprovider/types/types.go:92]: Can you please move this to https://github.com/kubermatic/machine-controller/blob/master/pkg/cloudprovider/util/net.go

File: pkg/cloudprovider/provider/aws/provider.go [hunk 2]
```diff
@@ -21,6 +21,7 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"github.com/kubermatic/machine-controller/pkg/cloudprovider/util"
```
Comment (moadqassem) [inline pkg/cloudprovider/provider/aws/provider.go:24]: Please regroup the imports

File: pkg/cloudprovider/util/net.go [hunk 3]
```diff
@@ -19,6 +19,7 @@ package util
 import (
 	"errors"
 	"fmt"
+	net2 "k8s.io/utils/net"
```
Comment (moadqassem) [inline pkg/cloudprovider/util/net.go:22]: Please regroup the imports

**Review by moadqassem — COMMENTED** [review]
@PratikDeoghare left some minor comments, otherwise lgtm

File: pkg/cloudprovider/provider/aws/provider.go (hunk 2 above)
Comment (PratikDeoghare) [reply pkg/cloudprovider/provider/aws/provider.go]: Linter accepts it :wink:

**Review by moadqassem — APPROVED** [review]
/approve
/lgtm

**Pull request merged**